ledger-api/grpc-definitions/com/digitalasset/admin/api/v1/package_management_service.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.admin.api.v1;
+
+option java_outer_classname = "PackageManagementServiceOuterClass";
+option java_package = "com.digitalasset.amin.api.v1";
+
+// Allows clients to query the DAML-LF packages that are supported by the server.
+service PackageManagementService {
+
+  // Returns the details of all supported packages.
+  rpc ListPackageDetails (ListPackageDetailsRequest) returns (ListPackageDetailsResponse);
+
+  // Uploads a DAR archive.
+  rpc UploadArchive (UploadArchiveRequest) returns (UploadArchiveResponse);
+
+}
+
+message ListPackageDetailsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
```

Comment:
as discussed it's a bit unclear whether ledger_id belongs to this API (ledger identifier is really just identifies what the participants index database incarnation, that is, it ties the ledger offsets into the concrete incarnation of the database and if it gets rebuilt the ledger identifier has to change).

Reply:
It is a tough question. It is easier to answer if this api moves into ledger.api.v0.admin package. The effects of this api are visible in the services of the ledger api. In some implementations, the data is stored on the ledger interspersed with the transactions.

Reply:
I think it's valid to have it here. The `ledger_id` in GSL was hashed into the genesis block and packages were published and evidenced on the ledger.